cmd/controller-manager/app/controller_manager.go

```diff
@@ -293,58 +293,70 @@ func StartControllers(s *options.ControllerManagerServer,
 	recorder record.EventRecorder,
 	stop <-chan struct{}) error {
 
-	// Get available service-catalog resources
-	glog.V(5).Info("Getting available resources")
-	availableResources, err := getAvailableResources(serviceCatalogClientBuilder)
-	if err != nil {
-		return err
+	// When Catalog Controller and Catalog API Server are started at the
+	// same time with API Aggregation enabled, it may take some time before
+	// Catalog registration shows up in API Server.  Re-query several times if
+	// necessary.
+
+	const maxAttempts = 4
+	for attempt := 1; ; attempt++ {
```

Comment:
you should use `wait.ExponentialBackoff` here instead, see: https://github.com/kubernetes-incubator/service-catalog/blob/master/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go#L192